modules/library/main/src/test/java/org/geotools/filter/visitor/SimplifyingFilterVisitorTest.java

```diff
@@ -224,8 +228,8 @@ public void testNegateBetween() {
         Literal l20 = ff.literal(20);
         Filter f = ff.not(ff.between(prop, l10, l20));
         Filter result = (Filter) f.accept(visitor, null);
-        assertEquals(ff.or(Arrays.asList((Filter) ff.less(prop, l10), ff.greater(prop, l20))),
-                result);
+        // *not* simplified
+        assertEquals(f, result);
```

Comment:
@aaime it is not a regression to remove a feature that does not work.

Suppose the property has values [5, 15, 25]. The pseudocode filter not(any-between(10, 20)) evaluates to false. Simplification converts this to or(any-less-than(10), any-greater-than(20)), which evaluates to true. Simplification is thus unvalid for multivalued properties.

The problem is that, to convert "between" to an "or" of "less-than" and "greater-than" requires consideration of *which* property values each child filter matches, which is not supported by the filter API. To do this would require something like a match-index array of booleans as the returned value, as in NumPy, and an element-wise "or".

I tried adding a (featureType instanceof SimpleFeatureType) guard but featureType is null for this test. I have not thought of a solution that works with the current filter API and does not break multivalued properties.